ext/ffi/ffi.c

```diff
@@ -5392,6 +5392,11 @@ static zend_result ffi_fixup_temporaries(void) {
 		++zend_ffi_cast_fn.T;
 		++zend_ffi_type_fn.T;
 	}
+#if !ZTS
+	ZEND_MAP_PTR(zend_ffi_new_fn.run_time_cache) = ZEND_MAP_PTR(((zend_internal_function *)zend_hash_str_find_ptr(&zend_ffi_ce->function_table, "new", sizeof("new")-1))->run_time_cache);
+	ZEND_MAP_PTR(zend_ffi_cast_fn.run_time_cache) = ZEND_MAP_PTR(((zend_internal_function *)zend_hash_str_find_ptr(&zend_ffi_ce->function_table, "cast", sizeof("cast")-1))->run_time_cache);
+	ZEND_MAP_PTR(zend_ffi_type_fn.run_time_cache) = ZEND_MAP_PTR(((zend_internal_function *)zend_hash_str_find_ptr(&zend_ffi_ce->function_table, "type", sizeof("type")-1))->run_time_cache);
+#endif
```

Comment:
Why do we need this?
Why this is done only for non-ZTS build?
Will we have any problems with ZTS build?

Please use ``ZEND_MAP_PTR_INIT`` instead of ``ZEND_MAP_PTR()`` on left size.

Reply:
I'm using MAP_PTR_INIT for everything where I assign a _value_, but in this case I want to copy whatever binary data is, thus I found using ZEND_MAP_PTR directly more fitting and symmetric with the right side.

Reply:
Also, only need it in NTS, as in ZTS this uses a proper map_ptr with is initialized when the function is created and thus existing when copied.
In NTS the pointer is initialized later in the startup sequence and thus needs to be copied back.